server/sonar-web/src/main/js/apps/component-measures/home/actions.js

```diff
@@ -59,11 +67,15 @@ export function fetchMeasures() {
         })
         .filter(measure => {
           const hasValue = measure.value != null;
-          const hasLeakValue = !!leakPeriod && measure.leak != null;
+          const hasLeakValue = measure.leak != null;
           return hasValue || hasLeakValue;
         });
 
-      dispatch(receiveMeasures(measures, r.periods));
+      const newBugs = measures.find(measure => measure.metric.key === 'new_bugs');
+
+      const periods = component.qualifier === 'APP' ? (newBugs ? [{ index: 1 }] : []) : r.periods;
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Extract this nested ternary operation into an independent statement. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=javascript%3AS3358)